presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -575,13 +581,27 @@ public boolean transitionToFinishing()
 
     private boolean transitionToFinished()
     {
+        try {
+            metadata.cleanupQuery(session);
```

Comment:
In the previous commit, cleanup was an async call. Why the change to synchronous?